### DISCUSSION TITLE
Welcome to GECV Discussions!

### DISCUSSION THREAD

**Discussion by HaoJun0823** [HaoJun0823/GECV#16] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by hacxerthanblack** [comment]
Is it possible to edit in game model, like higher quality mesh, texture etc ?

**Comment by HaoJun0823** [comment]
> Is it possible to edit in game model, like higher quality mesh, texture etc ?



> Is it possible to edit in game model, like higher quality mesh, texture etc ?

Yes, but the editing process was near hell.
If you want high definition, I recommend super resolution, refer to RTX Remix or NVIDIA Image Scaling. The memory allocation of the game cannot allocate too many resources.
If you want to do something else, I don’t have anything more to say at the moment because we are also experimenting.

**Comment by nachotacos69** [comment]
i manage to get some model from the dlc (the decryption method for the dlc of GE2 psp took me time)
![image](https://github.com/HaoJun0823/GECV/assets/99103531/39ed1bdf-4209-44e3-9a06-3ba110cfe46b)

**Reply by DarkGD123** [reply]
you have plans to create a tool for GE2 (psp) @nachotacos69, just asking though.. not forcing

**Reply by nachotacos69** [reply]
@DarkGD123 for now.. no, i dont have the will to do tons of things solo... so i only do helping and other stuff
 
:D

**Reply by nachotacos69** [reply]
I'm actually trying to recreate qpckeater for some .rdp/.edat files for older GE I guess @DarkGD123 but I don't think I'm have time to revise the entire code 💀

**Comment by nachotacos69** [comment]
here ya go @DarkGD123, use this tool that i made to mess around with GE2 (PSP) since you really want to mod or do whatever you want to do like exploration, etc. [its called a BLZ2 Roulette](https://github.com/nachotacos69/WikiEater/tree/main/GOD%20EATER%202/BLZ2Finder), the executable is already there.. so just read the instructions in the executable

Do note that im not involved/collaborating with HaoJun (no offense though to be honest), since he is busy with something else on GE PC related things so yeah..

**Reply by nachotacos69** [reply]
update: although attempt to replace any text isnt possible yet

**Comment by nachotacos69** [comment]
after some days fiddling around with the dlc, i manage to piece together kyuubi (magatsu)

![image](https://github.com/HaoJun0823/GECV/assets/99103531/66369508-cac0-4676-9954-d5e67a573e97)

**Comment by DarkGD123** [comment]
does anyone know how to gather riff data? (PSV/PSP)

**Reply by nachotacos69** [reply]
hmm, its hard but check the GECV-OLD repository for RIFF audios for vita

as for PSP.. well i don't know if @HaoJun0823 knows about .at3 formats. but i could explain things from here on then

from my experience on custom BGM's, in RIFF's headers somewhere in block 4 to 7, which is a uint32 value, you can use this to get the .at3 end offsets

- getting the .at3 end offset
![Screenshot (171)](https://github.com/HaoJun0823/GECV/assets/99103531/1d081b82-6d79-4bbf-aa8e-8707df0dce3d)

- using the value to go to the end offset of the file
![Screenshot (172)](https://github.com/HaoJun0823/GECV/assets/99103531/2e4e9ff1-a998-48fd-9bba-26b1a49adf77)


let me know if i did something stupid/incorrect (i can really tell if i did something wrong)

**Comment by DarkGD123** [comment]
So.. just checked 510's bilibili

like it's possible for GE2 to be translated, well it seems like it isn't 1.40

**Reply by nachotacos69** [reply]
i would rather not talk about this one since currently there's no fully info on this one

**Comment by DarkGD123** [comment]
Do you have a copy of GE2 (ISO+DLC) @nachotacos69

**Comment by DarkGD123** [comment]
@HaoJun0823 @nachotacos69 

anyone got any datas for PRES structure for GE2 (like the vanilla/portable)

current i only know about in the GEB variant which was provided in GEBCS code, which is right here https://github.com/gil-unx/GEBCS/tree/main/src

**Comment by nachotacos69** [comment]
@DarkGD123 
my own perspective or theories [i'll mention the GAME ID here too in case it needed for someone to double check the game]:

1. For God Eater [ULJS00237] and Burst [NPJHS50352]
- the use the same magic? "50 72 65 73 **[10]**", the brackets seems like a important value?
- the brackets seems to either define the PRES version or any other things. its value is `16` **based on u/int8 - u/int32** but in **u/int64** it represent as `30064771088` *im pretty sure u/int64 isnt required since the game only handle 32 bit stuff*
![Screenshot (210)](https://github.com/HaoJun0823/GECV/assets/99103531/1816d0d5-5850-481c-a64e-53d2a350ffb8)

2. in GOD EATER 2 [NPJH50832]
- it seems to use the same magic but now has a **[20]** instead of **[16]**. although it has space on your view
- Its value is `32` based on **uint8 - uint32** but in **u/tin64**, it represent as `2585632468958707744` *pretty big and im sure that this is not needed to unpack PRES*
![Screenshot (211)](https://github.com/HaoJun0823/GECV/assets/99103531/4dc39436-13ff-423a-bafa-8ef426a66315)

do note that i currently have no knowledge on fully extracting PRES data on both games especially in number 2 @DarkGD123 

feel free to correct my statement, im not always great at this kind of things but I'm always willing to help

**Comment by DarkGD123** [comment]
hello again fellas, what are you guys doing today?

**Reply by nachotacos69** [reply]
just doing stuff i guess, everyone has been busy with other projects

**Reply by DarkGD123** [reply]
is this accurate? [WikiEater - BLZ2](https://github.com/nachotacos69/WikiEater/blob/main/GOD%20EATER%202/BLZ2.md)

considering the structure text is chinese

**Reply by nachotacos69** [reply]
well. i made it as accurately as possible, i had to use deepL, and my brain to reconstruct into a accurate translation as possible

**Reply by nachotacos69** [reply]
well, considering GOD EATER 2 is made of C/C++ 32bit or any other code, because of the EBOOT.BIN information, so making a PRX plugin out of this just to extract blz2 files seems to be easy in mind but the problem/execution would be hard since that i don't have a clean computer/laptop to setup a ubuntu environment, since i had to do many things, basically like installing MIPS related code, Allegrex, and some more allegrex compiling and such, which overall takes ages

**Reply by DarkGD123** [reply]
hmm, i see.. well, can you share a code sample to make the blz2 file readable?

**Reply by nachotacos69** [reply]
![def](https://github.com/user-attachments/assets/0a8854cd-f892-463f-9c24-c9e95bc17f5a)

I believe this was the code, this is from my old **BLZ2BatchDeflate**, i got this from [BLZ2Utils](https://github.com/HaoJun0823/GECV-OLD/blob/74a9b3ba5ffb675f015f891f1fa3811916a965c6/GECV%20II/GECV_EX/Utils/BLZ2Utils.cs#L56) in the source code of GECV, the rest of the code of the batch is to extract the entire blz2 files, if its finished extracting one, it moves onto another

**Reply by DarkGD123** [reply]
i see..


***So... Do you have plans to make a GOD EATER 2 tool?***

**Reply by nachotacos69** [reply]
No. too many roadblocks for me

@HaoJun0823 has been working on his own tool but slowly, as Cody told me

if i would make mine though, the roadblock for me is the `EBOOT.BIN`
since i need to modify the fixed buffer of the character limitations which is 10 (based on @gil-unx answer)
which also needed to be decompiled through ASM, which i am not good at

**Reply by nachotacos69** [reply]
Late continuation response: but yeah, all I could say is maybe soon the older games will have their own tools and such respectfully. But only time can tell

**Reply by DarkGD123** [reply]
I see

**Comment by DarkGD123** [comment]
old stuff @nachotacos69 from quickbms
filename: `god_eater_2.bms`

```
# God Eater 2 (script 0.2.3a)
#   Experimental, it's a scanner.
# script for QuickBMS http://quickbms.aluigi.org

comtype deflate
math FLAGS_BITS = 4

math LAST_OFF = 0   # used by the scanner
set PATH string ""  # used by the scanner

set SIGN_QPCK binary "\x58\x28\x40\x37"

getdstring SIGN 4
goto 0
if SIGN u== "Pres"
    callfunction EXTRACT_PRES 1

elif SIGN u== "rdp "
    getdstring SIGN 16
    getdstring DUMMY 16

        #
        # THIS IS A SCANNER! I don't have the game and I don't know where is the index file!
        #

    get RDP_SIZE asize
    for FILE_NUM = 0
        padding 0x800

        savepos OFFSET
        if OFFSET == RDP_SIZE
            break
        endif

        math LAST_OFF = -1  # SIZE is not known and so we have to guess/scan
        callfunction EXTRACT_FILE 1

        goto LAST_OFF
    next FILE_NUM

elif SIGN u== SIGN_QPCK

    get DUMMY long  # 0x37402858
    get FILES long
    for i = 0 < FILES
        get OFFSET longlong
        get NAME_HASH longlong
        get SIZE long

        xmath LAST_OFF "OFFSET + SIZE"
        savepos BCK_OFF
        goto OFFSET

        math FILE_NUM = i   # compatible with EXTRACT_FILE
        callfunction EXTRACT_FILE # do not use 1 here
        goto BCK_OFF
    next i

else
    print "Error: unknown file type %SIGN%"
    cleanexit
endif



startfunction EXTRACT_FILE
    set PATH string FILE_NUM
    #string NAME p "%s/" PATH
    set NAME string ""

    getdstring SIGN 4
    set SIGN3 string SIGN
    string SIGN3 - 1
    goto OFFSET

    if SIGN u== "Pres"
        callfunction EXTRACT_PRES 1

    elif SIGN3 u== "blz"
        getdstring SKIP 0x10    # blz4
        if LAST_OFF == -1
            findloc LAST_OFF binary "\0\0\0\0"
            findloc TMP binary SIGN # it's the full blz*
            if TMP u< LAST_OFF
                math LAST_OFF = TMP
                math LAST_OFF - 2   # ???
            endif
            xmath SIZE "LAST_OFF - OFFSET"
        endif
        callfunction UNPACK_BLZ

    elif SIGN u== "RIFF"
        getdstring SIGN 4
        get SIZE long
        math SIZE + 8
        log "" OFFSET SIZE
        xmath LAST_OFF "OFFSET + SIZE"

    else
        #print "Error: unknown scanned signature %SIGN%"
        #cleanexit

        if LAST_OFF == -1
            findloc LAST_OFF binary "Pres"
            xmath SIZE "LAST_OFF - OFFSET"
        endif
        log "" OFFSET SIZE
    endif
endfunction


startfunction EXTRACT_PRES
    set NAME string ""
    set FOLDER string ""
    set EXT string ""

    savepos PRES_OFF
    idstring "Pres"
    get DUMMY1 long
    get DUMMY2 long
    get DUMMY3 long
    get DATA_OFF long
    get ZERO long
    get ZERO long
    get SET_OF_FILES long    # I don't know how to call this number
    if SET_OF_FILES <= 1
        savepos OFFSET
        math OFFSET - PRES_OFF 
        putarray 0 0 OFFSET
        math SET_OF_FILES = 1   # necessary or it will not work!
    else
        for SET_OF_FILES_IDX = 0 < SET_OF_FILES
            get OFFSET long
            get DUMMY long
            putarray 0 SET_OF_FILES_IDX OFFSET
        next SET_OF_FILES_IDX
    endif
    for SET_OF_FILES_IDX = 0 < SET_OF_FILES
        getarray OFFSET 0 SET_OF_FILES_IDX
        math OFFSET + PRES_OFF
        goto OFFSET

        get NAMES_OFF long
        get NAMES_ELEMENTS long
        get DUMMY_OFF long
        get DUMMY long
        get INFO_OFF long
        get FILES long
        get DUMMY long
        get DUMMY long
        get DUMMY long
        get DUMMY long
        get DUMMY_OFF long
        get DUMMY long

        savepos LAST_OFF

        math INFO_OFF + PRES_OFF
        for i = 0 < FILES
            goto INFO_OFF
            get OFFSET long
            get SIZE long
            get NAME_OFF long
            get NAME_ELEMENTS long
            get DUMMYe long
            get DUMMYf long
            get DUMMYg long
            get XSIZE long      # uncompressed size
            savepos INFO_OFF

            if OFFSET & 0x80000000  # there is also 0x40000000 but it will return less extracted files

                xmath FLAGS "OFFSET >> FLAGS_BITS"
                xmath OFFSET "OFFSET & ((1 << (32 - FLAGS_BITS)) - 1)"

                math NAME_OFF + PRES_OFF
                goto NAME_OFF
                get NAME_OFF long
                get EXT_OFF long
                get FOLDER_OFF long

                if NAME_ELEMENTS >= 1
                    math NAME_OFF + PRES_OFF
                    goto NAME_OFF
                    get NAME string
                    if NAME_ELEMENTS >= 2
                        math EXT_OFF + PRES_OFF
                        goto EXT_OFF
                        get EXT string
                        if NAME_ELEMENTS >= 3
                            math FOLDER_OFF + PRES_OFF
                            goto FOLDER_OFF
                            get FOLDER string
                        endif
                    endif
                endif

                #string NAME p "%s/%s.%s" FOLDER NAME EXT
                string NAME p "%s/%s/%s.%s" PATH FOLDER NAME EXT

                math OFFSET + PRES_OFF

                xmath TMP "OFFSET + SIZE"
                if TMP u> LAST_OFF
                    math LAST_OFF = TMP
                endif

                if SIZE == XSIZE
                    log NAME OFFSET SIZE
                else
                    callfunction UNPACK_BLZ
                endif

            endif
        next i
    next SET_OF_FILES_IDX
endfunction

startfunction UNPACK_BLZ
    savepos BACKUP_OFF
    goto OFFSET
    getdstring SIGN3 3
    if SIGN3 u== "blz"
        math UNCOMPRESSED_SIZE = -1
        get BLZ_VER byte
        if BLZ_VER u== '4'
            comtype zlib
            get UNCOMPRESSED_SIZE long
            get ZERO long
            get ZERO long
            getdstring DUMMY 0x10
        elif BLZ_VER u== '2'
            comtype deflate
        else
            print "Error: unknown blz %BLZ_VER%, contact me"
            cleanexit
        endif
        xmath OFFSET_LIMIT "(OFFSET + SIZE) - 15"   # the "minus" is a work-around
        log MEMORY_FILE 0 0
        append
        do
            get CHUNK_ZSIZE short
            savepos CHUNK_OFF
            if CHUNK_ZSIZE == 0
                log MEMORY_FILE CHUNK_OFF 0x10000
            else
                clog MEMORY_FILE CHUNK_OFF CHUNK_ZSIZE 0x10000
            endif
            math CHUNK_OFF + CHUNK_ZSIZE
            goto CHUNK_OFF
            get MEM_SIZE asize MEMORY_FILE
        while CHUNK_OFF u< OFFSET_LIMIT
        append
        get SIZE asize MEMORY_FILE
        log NAME 0 SIZE MEMORY_FILE
    else
        log NAME OFFSET SIZE
    endif
    goto BACKUP_OFF
endfunction
```

**Comment by DarkGD123** [comment]
There's hasn't been any update @nachotacos69, got any ideas?

**Comment by nachotacos69** [comment]
3 and a half months of full silence, yeah they are busy. So might as well have patience. Even though I haven't heard or gotten any updates for their new projects or other things at all.

**Comment by nachotacos69** [comment]
Good Morning, @HaoJun0823 
if you're still busy at **GOD EATER 2** or not active on it anymore, i manage to get in contact with SkyBladeCloud. Who tried to make a tool for **GOD EATER 2** 7 years ago.

Here's the code he gave to me.
[FileInfo.txt](https://github.com/user-attachments/files/17106255/FileInfo.txt)
[PRES.txt](https://github.com/user-attachments/files/17106256/PRES.txt)
[EntryGroup.txt](https://github.com/user-attachments/files/17106257/EntryGroup.txt)
[FileData.txt](https://github.com/user-attachments/files/17106259/FileData.txt)
[FileEntry.txt](https://github.com/user-attachments/files/17106260/FileEntry.txt)

Comment:
![skyblade](https://github.com/user-attachments/assets/7c6f095e-21ae-404b-aa51-620607680939)

**Reply by nachotacos69** [reply]
there's still missing parts here and there but its a reference after all, so yeah

**Reply by nachotacos69** [reply]
i managed to replace the empty `BLZ2.Decompress` with the BLZ2 unpacking method from your source code: https://github.com/HaoJun0823/GECV-OLD/blob/master/GECV%20II/GECV_EX/Utils/BLZ2Utils.cs

**Reply by HaoJun0823** [reply]
> Good Morning, @HaoJun0823 if you're still busy at **GOD EATER 2** or not active on it anymore, i manage to get in contact with SkyBladeCloud. Who tried to make a tool for **GOD EATER 2** 7 years ago.
> 
> Here's the code he gave to me. [FileInfo.txt](https://github.com/user-attachments/files/17106255/FileInfo.txt) [PRES.txt](https://github.com/user-attachments/files/17106256/PRES.txt) [EntryGroup.txt](https://github.com/user-attachments/files/17106257/EntryGroup.txt) [FileData.txt](https://github.com/user-attachments/files/17106259/FileData.txt) [FileEntry.txt](https://github.com/user-attachments/files/17106260/FileEntry.txt)
> 
> Comment: ![skyblade](https://private-user-images.githubusercontent.com/99103531/370087256-7c6f095e-21ae-404b-aa51-620607680939.PNG?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjcxNDA5NjEsIm5iZiI6MTcyNzE0MDY2MSwicGF0aCI6Ii85OTEwMzUzMS8zNzAwODcyNTYtN2M2ZjA5NWUtMjFhZS00MDRiLWFhNTEtNjIwNjA3NjgwOTM5LlBORz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA5MjQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwOTI0VDAxMTc0MVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTZmOWQyZWEwMTA0ZjQwNjliMDQ2NTg1Zjc4ZWRiM2Y5M2ZjZGM1ODM2OWVlMGEyZGJmZTJmN2RmN2ZkOWU2NDEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.yaEQEIUWAwU-RdZciXFOqjMKkUeozQfebA8XbAPfMXI)

This information is all known.
For now, we are not making any new content because the PC version has already done everything.

Regarding PSP, PSV, and PS4, these are no longer in our plans, I have reiterated this many times.

**Comment by DarkGD123** [comment]
yo, good evening @HaoJun0823, where do you find the Radio chatters?

**Reply by nachotacos69** [reply]
the radio chatter strings?

**Comment by nachotacos69** [comment]
<img width="1222" height="770" alt="image" src="https://github.com/user-attachments/assets/44a83c6d-b585-4f76-80e0-cedfb9974542" />
<img width="1219" height="778" alt="image" src="https://github.com/user-attachments/assets/e9d05c37-2eb5-4e59-960d-8d40a0c2d06c" />

i've been trying to understand KST file for a while now. but yeah. i'll just take time understanding the KST itself.
